### PR TITLE
Remove RedirectURL parameter from auth code flow

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
To align with the MSAL implementation which relies on the native client
redirect URL.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/azidentity/aad_identity_client.go [hunk 1]
```diff
@@ -415,14 +415,15 @@ func (c *aadIdentityClient) authenticateAuthCode(ctx context.Context, tenantID s
...
+func (c *aadIdentityClient) createAuthorizationCodeAuthRequest(ctx context.Context, tenantID string, clientID string, authCode string, clientSecret string, scopes []string) (*azcore.Request, error) {
 	data := url.Values{}
 	data.Set(qpGrantType, "authorization_code")
 	data.Set(qpClientID, clientID)
 	if clientSecret != "" {
 		data.Set(qpClientSecret, clientSecret) // only for web apps
 	}
-	data.Set(qpRedirectURI, redirectURI)
+	// similar to MSAL, we use a hard-coded redirect URI here as we never actually redirect to it
+	data.Set(qpRedirectURI, "https://login.microsoftonline.com/common/oauth2/nativeclient")
```
Comment (catalinaperalta) [inline sdk/azidentity/aad_identity_client.go:426]: I'm just curious if you know about any app type that might need a custom redirect URI? Or if the MSAL team ever mentioned any? Not blocking on it, but just curious about this.

**Review by catalinaperalta — APPROVED** [review]
LGTM

File: sdk/azidentity/aad_identity_client.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azidentity/aad_identity_client.go:426]: In our [conversation](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/68#discussion_r517352378) the MSAL folks said that "MSAL stops the redirect" so there's no need to actually perform the redirect.

If we do end up needing this we can probably add it to the options struct given that it's optional from an MSAL perspective.

Comment (catalinaperalta) [reply sdk/azidentity/aad_identity_client.go:426]: Ok that sounds like a good plan to me.

**Comment by harikb** [comment]
This change is causing a problem for me.  The "redirectURI" parameter is needed in the flow of "Confidential client's Auth Code" flow as well.

If you look at your sample code https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/test/devapps/confidential_auth_code_sample.go 

redirectToURLConfidential in the sample needs to use an app specific redirect URL, because this is the one really sent to the browser. Later when 'code' is given to the site and we try to fetch an access token, that code no longer works because the above change hard-codes a particular  URL of 

`https://login.microsoftonline.com/common/oauth2/nativeclient`

@jhendrixMSFT  @catalinaperalta

**Pull request merged**